features/buyers/accounts/new.feature

```diff
@@ -37,3 +37,22 @@ Feature: Audience > Accounts > New
     And field "Email" has inline error "should look like an email address"
     And field "Organization/Group Name" has inline error "can't be blank"
     But field "Password" has no inline error
+
+  @javascript
+  Scenario: Create account with fields with choices
+    Given the provider has the following fields defined for accounts:
+      | name                 | choices                 | label                  | required | read_only | hidden |
+      | country              |                         | Country                | true     |           |        |
+      | field_with_choices   | hello, option1, option2 | Fields with choices    | true     |           |        |
+    And they go to the new buyer account page
+
+    When they select "Spain" from "Country"
+    And they select "option1" from "Fields with choices"
+    And the form is submitted with:
+      | Username                | alice              |
+      | Email                   | alice@example.com  |
+      | Organization/Group Name | Alice's Web Widget |
```

Comment:
```suggestion
    When the form is submitted with:
      | Username                | alice              |
      | Email                   | alice@example.com  |
      | Organization/Group Name | Alice's Web Widget |
      | Country | Spain |
      | Fields with choices | option1 |
```

Reply:
That's what I was considering at first, but that doesn't verify that the Country and Fields with choices is a **select** field. The issue specifically was that it was rendered as an input, and that was wrong, so we need to verify that after the fix these fields look exactly as expected.

Reply:
If/when we change the field, we'll change the test.

I understand that the tests should be generic, but if they are too generic to miss the bugs introduced inadvertently, they are pretty useless then.

Reply:
My point is verifying that "Country" and "Fields with choices" are a `<select>` should be a unit test for `PatternflyInputInput#input` or perhaps `Fields::PatternflyFormBuilder#output_html` and this scenario should test a user updating an account, regardless of implementation.

To test the implementation here, I would make different assertions (one test the html, another submits the form):

```suggestion
    And there is a select "Country"
    And there is a select "Fields with choices"
    When the form is submitted with:
      | Username                | alice              |
      | Email                   | alice@example.com  |
      | Organization/Group Name | Alice's Web Widget |
      | Country | Spain |
      | Fields with choices | option1 |
```

This at least helps differentiating aspects/concepts in a very specific scenario and let "form submitted" do its job properly. This step should be usable with every single form.